src/ghostfat.c

```diff
@@ -266,31 +306,49 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
     // second FAT is same as the first...
     if ( sectionRelativeSector >= BPB_SECTORS_PER_FAT ) sectionRelativeSector -= BPB_SECTORS_PER_FAT;
 
+    uint16_t* data16 = (uint16_t*) (void*) data;
+
     if ( sectionRelativeSector == 0 )
     {
       // first FAT entry must match BPB MediaDescriptor
       data[0] = BPB_MEDIA_DESCRIPTOR_BYTE;
+      data[1] = 0xff;
+
+      // cluster 1 is reserved
+      data16[1] = 0xffff;
+
       // WARNING -- code presumes only one NULL .content for .UF2 file
-      //            and all non-NULL .content fit in one sector
       //            and requires it be the last element of the array
-      uint32_t const end = (NUM_FILES * FAT_ENTRY_SIZE) + (2 * FAT_ENTRY_SIZE);
-      for ( uint32_t i = 1; i < end; ++i ) data[i] = 0xff;
+      uint32_t v = 2;
+      for (uint32_t i=0; i<NUM_FILES-1; i++)
```

Comment:
Two issues exist in the code as written:

1. Assumption that FAT chain for all files (other than CURRENT.UF2) will be in first sector of FAT
2. No `static_assert` to ensure that this condition is provably true.

Reply:
resolved in PR #187